azure-pipelines.yml

```diff
@@ -51,3 +51,39 @@ jobs:
 
   steps:
   - template: ci/azure-steps.yml
+
+- job: msys2_mingw
+  pool:
+    vmImage: VS2017-Win2016
+  strategy:
+    matrix:
+      i686:
+        MSYSTEM: MINGW32
+        MSYS2_ARCH: i686
+      x86_64:
+        MSYSTEM: MINGW64
+        MSYS2_ARCH: x86_64
+  steps:
+    - script: git clone https://github.com/lazka/msys2-ci-base.git %CD:~0,2%\msys64
+      displayName: Install MSYS2
+    - script: |
+        set PATH=%CD:~0,2%\msys64\usr\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem
+        %CD:~0,2%\msys64\usr\bin\pacman --noconfirm -Syyuu
+      displayName: Update MSYS2
+    - script: |
+        set PATH=%CD:~0,2%\msys64\usr\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem
+        %CD:~0,2%\msys64\usr\bin\pacman --noconfirm --needed -S git base-devel mingw-w64-$(MSYS2_ARCH)-toolchain mingw-w64-$(MSYS2_ARCH)-python3 mingw-w64-$(MSYS2_ARCH)-python3-setuptools mingw-w64-$(MSYS2_ARCH)-python2 mercurial
+      displayName: Install Dependencies
+    - script: |
+        set PATH=C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem
+        set CHERE_INVOKING=yes
```

Comment:
This could appear in a `variables:` block inside the job, instead of as code

Reply:
what about adding it to "env:" for this step instead?